src/Arcus.Security.Providers.KeyPerFile/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Extensions.Configuration.KeyPerFile;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+
+namespace Arcus.Security.Providers.KeyPerFile
+{
+    public static class SecretStoreBuilderExtensions
+    {
+        public static SecretStoreBuilder AddKeyPerFile(this SecretStoreBuilder builder, string directoryPath, bool optional = true)
```

Comment:
Should we use the `optional` here? It was supposed to be used (I think) to return `null` instead of throwing, but because we use the secret store, this feature is resulting in the same result bc the secret store will eventuall throw if the secret was not found in any of the registered providers.

Reply:
If optional is set to false, an exception will be thrown when the directory-path (where the secrets are located inside the container) doesn't exist.
If set to true, no exception will be thrown at that point, however, an exception will be thrown later (which says that the secret could not be found).   However, if we have multiple secretstores configured, the secret might as well be found by another secret provider ....
Don't know what the best approach is here, therefore I've just taken over the functionality that exists in the 'KeyPerFile' config provider.

Reply:
If it's just for the file then I would use false internally because otherwise there is a "setup issue", but I might be too extreme?

Reply:
Yes, I agree with @tomkerkhove , our other providers are also throwing on 'setup' things and if we would want to have such an optional functionality, I think we should look into it on a broader level and how we can use it for every provider. So, for now, maybe set it on `false` and we can maybe look at it in an issue later.

Reply:
I've changed and pushed this, but would like to do some additional testing.

Reply:
No problem, let us know when you are ready!

Reply:
I've changed this, but now this means that if you specify a non-existing directory, the container will fail to start.  That is a good example of failing fast but I don't know if you really want that.

What if you have a program which has multiple secret providers configured.   In some cases, your program might not be running in a container and needs to retrieve secrets from another location.  In that case, you still need to have that directory present.